addons/website_sale_slides/views/website_sale_templates.xml

```diff
@@ -21,16 +21,18 @@
                     <span t-if="course.image_1920" t-field="course.image_1920" t-options="{'widget': 'image', 'class': 'my-2'}"/>
                     <img t-else="" class="img img-fluid my-2" src="/website_slides/static/src/img/channel-training-default.jpg"/>
                 </div>
+                <t t-set="invitation_link" t-value="env['slide.channel.partner'].sudo().search([
+                    ('channel_id', '=', course.id), ('partner_id', '=', order.partner_id.id)]).invitation_link"/>
```

Comment:
Searching in the XML is rather dirty, especially considering it's done in a double for loop ^^

I think it would be better to do a clean override in the "_prepare_shop_payment_confirmation_values" method and add our membership_records (you can call it "course_memberships" or something similar) that holds the slide.channel.partner records of the courses associated with this order.

And then here you would check if it's available and filter on the course, with a fallback to the old "/slides/course" URL.

A bit more code, but a lot cleaner in the end.

What do you think?

Reply:
yes ok :) 

> And then here you would check if it's available and filter on the course, with a fallback to the old "/slides/course" URL.

I used a dictionnary (so no need to filter), and about the fallback there should be always a `slide.channel.partner` no (thanks to `_action_confirm`) ? (and if there's no `slide.channel.partner`, then it's a bug, and it's better to crash and to not ignore it imo, so we detect it quickly)